ptf/saidebugcounters.py

```diff
@@ -0,0 +1,2194 @@
+                'acl_table': acl_table_id,
+                'acl_entry': acl_entry,
+                'acl_table_group': acl_table_group,
+                'acl_group_member': acl_group_member})
+        return acl
+
+
+class GetDebugCounterEnumValuesCapabilities(BaseDebugCounterClass):
+    ''' SAI query DebugCounter enum values capabilities. '''
+
+    def runTest(self):
+        # Supported SAI IN drop reason list
+        in_caps_list = [
+            SAI_IN_DROP_REASON_INGRESS_VLAN_FILTER,
+            SAI_IN_DROP_REASON_L2_ANY,
+            SAI_IN_DROP_REASON_L3_ANY,
+            SAI_IN_DROP_REASON_SMAC_MULTICAST,
+            SAI_IN_DROP_REASON_TTL,
+            SAI_IN_DROP_REASON_IP_HEADER_ERROR,
+            SAI_IN_DROP_REASON_SIP_MC,
+            SAI_IN_DROP_REASON_IRIF_DISABLED,
+            SAI_IN_DROP_REASON_ACL_ANY,
+            SAI_IN_DROP_REASON_DIP_LOOPBACK,
+            SAI_IN_DROP_REASON_SIP_LOOPBACK,
+            SAI_IN_DROP_REASON_SIP_CLASS_E,
+            SAI_IN_DROP_REASON_DIP_LINK_LOCAL,
+            SAI_IN_DROP_REASON_SIP_LINK_LOCAL,
+            SAI_IN_DROP_REASON_SIP_UNSPECIFIED,
+            SAI_IN_DROP_REASON_UC_DIP_MC_DMAC,
+            SAI_IN_DROP_REASON_MPLS_MISS,
+            SAI_IN_DROP_REASON_SRV6_LOCAL_SID_DROP]
+
+        # Make sure SAI_IN_DROP_REASON_SMAC_EQUALS_DMAC is supported
+        # by current profile
+        if self.isInDropReasonSupported([SAI_IN_DROP_REASON_SMAC_EQUALS_DMAC]):
+            in_caps_list.append(SAI_IN_DROP_REASON_SMAC_EQUALS_DMAC)
+
+        in_dc_caps_list = sai_thrift_query_attribute_enum_values_capability(
+            self.client,
+            SAI_OBJECT_TYPE_DEBUG_COUNTER,
+            SAI_DEBUG_COUNTER_ATTR_IN_DROP_REASON_LIST)
+        print("Supported in drop_reasons: %s len=%d"
+              % (map_drop_reason_to_string(in_dc_caps_list),
+                 len(in_dc_caps_list)))
+        for drop_reason in in_caps_list:
+            found = False
+            for in_drop_reason in in_dc_caps_list:
+                if drop_reason == in_drop_reason:
+                    found = True
+            if found is False:
+                print("Unable to find drop reason: %s (%d)"
+                      % (map_drop_reason_to_string([drop_reason]),
+                         drop_reason))
+            self.assertTrue(found is True)
+        self.assertTrue(len(in_dc_caps_list) == len(in_caps_list))
+        print("\tok")
+
+        out_dc_cap_list = sai_thrift_query_attribute_enum_values_capability(
+            self.client,
+            SAI_OBJECT_TYPE_DEBUG_COUNTER,
+            SAI_DEBUG_COUNTER_ATTR_OUT_DROP_REASON_LIST)
+        print("Supported out drop_reasons: %s len=%d"
+              % (map_drop_reason_to_string(out_dc_cap_list),
+                 len(out_dc_cap_list)))
+        # Currently out drop reasons not supported
+        self.assertTrue(len(out_dc_cap_list) == 0)
+        print("\tok")
+
+
+class GetDebugCounterAvailability(BaseDebugCounterClass):
```

Comment:
It seems to me this test shows that we have some debug counter types that are unsupported. Is it worth to modify this test to hide this info?

Reply:
no, I think it works as designed